src/tools.js

```diff
@@ -1,18 +1,179 @@
 const Apify = require('apify');
-const camelcaseKeysRecursive = require('camelcase-keys-recursive');
 const moment = require('moment');
 const get = require('lodash.get');
+const csvToJson = require('csvtojson');
 const currencies = require('./currencyCodes.json');
 
-const { utils: { log } } = Apify;
+const { utils: { log, requestAsBrowser, sleep } } = Apify;
 // log.setLevel(log.LEVELS.DEBUG);
 const { callForReviews } = require('./api');
 const {
     HISTOGRAM_ITEMS_COUNT,
     MIN_LIMIT,
     MAX_LIMIT,
     DATE_FORMAT,
+    URL_WITH_ROOMS_REGEX,
+    DEFAULT_LIMIT_POINTS,
+    DEFAULT_TIMEOUT_MILLISECONDS,
+    DEFAULT_MIN_PRICE,
+    DEFAULT_MAX_PRICE,
+    DEFAULT_LOCALE,
 } = require('./constants');
+const { cityToAreas } = require('./mapApi');
+const { getLocale } = require('./localization');
+
+/**
+ * @param {Apify.Session | null} session
+ */
+const getRequestFnc = (session, proxy, locale = DEFAULT_LOCALE) => async (url, opts = {}) => {
+    const getData = async (attempt = 0) => {
+        let response;
+        const requestUrl = new URL(url);
+        requestUrl.searchParams.set('locale', locale);
+
+        const options = {
+            url: requestUrl.toString(),
+            json: false,
+            headers: {
+                'X-Airbnb-API-Key': process.env.API_KEY,
+            },
+            proxyUrl: proxy.newUrl(session ? session.id : `airbnb_${Math.floor(Math.random() * 100000000)}`),
```

Comment:
Why random session number? We can just put there underfined

Reply:
Didn't write this one but good to know 👍